docs/source/basics/defining_environment.md

```diff
@@ -0,0 +1,239 @@
+
+To understand the generated Dockefile, here is an example with instructive comments generated by running `latch init --template subprocess --dockerfile example_workflow`:
+
+```Dockerfile
+# latch base image + dependencies for latch SDK --- removing these will break the workflow
+from 812206152185.dkr.ecr.us-west-2.amazonaws.com/latch-base:ace9-main
+run pip install latch==2.12.1
+run mkdir /opt/latch
+
+# install system requirements
+copy system-requirements.txt /opt/latch/system-requirements.txt
+run apt-get update --yes && xargs apt-get install --yes </opt/latch/system-requirements.txt
+
+# copy all code from package (use .dockerignore to skip files)
+copy . /root/
+
+# set environment variables
+env BOWTIE2_INDEXES=reference
+
+# latch internal tagging system + expected root directory --- changing these lines will break the workflow
+arg tag
+env FLYTE_INTERNAL_IMAGE $tag
+workdir /root
+```
+
+Latch has three base images, one without hardware acceleration drivers, one with CUDA drivers, and one with OPENCL drivers. To use the CUDA or OPENCL base image, use the `--cuda` or `--opencl` flags when running `latch init`.
+
+## Ignoring files
+
+In the above Dockerfile, the line `copy . /root/` recursively copies all files and directories from the workflow directory into the Docker image. Even if files in the workflow directory are unused by the workflow, they will still be copied. Having unnecessary files in the image increases its size, slowing workflow execution and registration. Additionally, any changes to said files will cause image rebuilding when running `latch register` or `latch develop` on the workflow, making the process slower. To mitigate this, Latch uses a `.dockerignore` file in the workflow directory. This file should contain a list of files and directories that should be skipped when copying files into your container, as documented [here](https://docs.docker.com/engine/reference/builder/#dockerignore-file). Each workflow begins with a default `.dockerignore` file that ignores common pesky files (for example, `.latch_report.tar.gz` and other auto-generated files).
+
+## Docker Limitations
+
+The difference between a latch task environment and running code on a Linux machine is that we restrict access to root system resources. For example, `/dev` and the networking stack are restricted, so creating mounts using `/dev/fuse` is not permitted. We limit this behavior to prevent users from accessing sensitive system resources that could influence other tasks running on the same machine. In the future, we will support full container isolation, allowing users to treat their containers as complete linux machines.
```

Comment:
no mounts will work on normal docker (like the `mount` command itself will error)
also some parts will work under sysbox